test/rules/no-sparse-arrays/test.ts.lint

```diff
@@ -0,0 +1,22 @@
+[1, , 3];
+   ~ [0]
+
+// Destructuring allowed.
+const [foo, , bar] = [];
+const [{foo: [, bar]}] = [];
+const [foo = [, 1]] = [];
+              ~ [0]
+
+[foo, , bar] = [];
+[{foo: [, bar]}] = [];
+[foo = [, 1]] = [];
```

Comment:
can you add another test case like: 
```ts
[foo = bar([,])] = [];
            ~[0]
```
I know this works with your current implementation. Just to make sure we won't break it in the future.